pkg/obs/services/buckets.go

```diff
@@ -0,0 +1,200 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package services
+
+import (
+	"fmt"
+	"github.com/huaweicloud/huaweicloud-csi-driver/pkg/config"
+	"github.com/huaweicloud/huaweicloud-sdk-go-obs/obs"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"net/http"
+)
+
+func GetBucket(c *config.CloudCredentials, bucketName string) (string, error) {
+	exists, err := CheckBucketExists(c, bucketName)
+	if err != nil {
+		return "", err
+	}
+	if !exists {
+		return "", nil
+	}
+	isParallelFile, err := CheckBucketFSStatus(c, bucketName)
+	if err != nil {
+		return "", err
+	}
+	if !isParallelFile {
+		return "", status.Errorf(codes.Unavailable, "Error, the OBS bucket %s is not a parallel file system", bucketName)
+	}
+	return bucketName, nil
+}
+
+func GetBucketMetadata(c *config.CloudCredentials, bucketName string) (*obs.GetBucketMetadataOutput, error) {
+	client, err := getObsClient(c)
+	if err != nil {
+		return nil, err
+	}
+	input := &obs.GetBucketMetadataInput{}
+	input.Bucket = bucketName
+	output, err := client.GetBucketMetadata(input)
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "Error, OBS instance %s does not exist, error: %v", bucketName, err)
+	}
+	return output, nil
+}
+
+func CheckBucketExists(c *config.CloudCredentials, bucketName string) (bool, error) {
+	client, err := getObsClient(c)
+	if err != nil {
+		return false, err
+	}
+	_, err = client.HeadBucket(bucketName)
+	if err == nil {
+		return true, nil
+	}
+	if obsError, ok := err.(obs.ObsError); ok {
+		if obsError.StatusCode == http.StatusNotFound {
+			return false, nil
+		}
+	}
+	return false, status.Errorf(codes.Internal, "Error, heading OBS instance %s, error: %v", bucketName, err)
+}
+
+func CheckBucketFSStatus(c *config.CloudCredentials, bucketName string) (bool, error) {
+	metadata, err := GetBucketMetadata(c, bucketName)
+	if err != nil {
+		return false, err
+	}
+	return metadata.FSStatus == obs.FSStatusEnabled, nil
+}
+
+func CreateBucket(c *config.CloudCredentials, bucketName string) (string, error) {
+	client, err := getObsClient(c)
+	if err != nil {
+		return "", err
+	}
+	input := &obs.CreateBucketInput{}
+	input.Bucket = bucketName
+	input.Location = c.Global.Region
+	input.ACL = obs.AclPrivate
+	input.IsFSFileInterface = true
+	_, err = client.CreateBucket(input)
+	if err == nil {
+		return bucketName, nil
+	}
+	if obsError, ok := err.(obs.ObsError); ok {
+		if obsError.StatusCode == http.StatusConflict {
+			return "", status.Errorf(codes.AlreadyExists, "Error, creating OBS instance %s existed, error: %v", bucketName, err)
```

Comment:
how about this:
if obsError, ok := err.(obs.ObsError); ok && obsError.StatusCode == http.StatusConflict{
      return "", status.Errorf(codes.AlreadyExists, "Error, creating OBS instance %s existed, error: %v", bucketName, err)
}